Add installer feature

mbacovsky opened the pull request:
The installer feature can
 - parse the answer files regardless of installer version
 - detect installer type (scenario, non-scenario based)
 - detect last used scenario
 - list related config files

mbacovsky commented:
I isolated the installer feature into separate PR. This feature will be needed in backup script and in the hammer feature refactoring.

iNecas commented on test/data/installer/capsule-installer/config_header.txt:
```diff
@@ -0,0 +1,8 @@
+# Format:
```
Nitpick: having `test/definitions/data` would be a better place for definitions-specific mock data IMO, but leaving the decision up to you

mbacovsky replied:
@iNecas I had it there originally. Then I added some tests for the hammer utils and while it called the feature internaly I had to mock the configs there so I moved the data to common space. 

Is it allowed practice to use definitions from the `lib`? For Hammer utils it made sense as a short term solution but could be avoided. Testing the other parts e.g. `cli` might also benefit from common data for mock ups. Saying that these data aren't that definition-specific so it makes sense to keep it shared among lib and def's tests.

iNecas replied:
:+1: then

iNecas commented:
The code works for me, @kgaikwad / @johnpmitsch any comments before merging?

kgaikwad commented:
@iNecas, 
For me looks good.

johnpmitsch commented on definitions/features/installer.rb:
```diff
@@ -0,0 +1,75 @@
+                        :scenarios
+                      elsif find_package('katello-installer')
+                        :katello
+                      elsif find_package('capsule-installer')
```
Upstream foreman-proxy-with-contents will use a `foreman-proxy-content` package, shouldn't this be accounted for? i.e.
```
[vagrant@bacon-proxy ~]$ rpm -qa | grep proxy
foreman-proxy-content-3.7.0-1.nightly.el7.noarch
bacon-proxy.jomitsch.example.com-apache-1.0-1.noarch
bacon-proxy.jomitsch.example.com-qpid-client-cert-1.0-1.noarch
gssproxy-0.7.0-4.el7.x86_64
bacon-proxy.jomitsch.example.com-puppet-client-1.0-1.noarch
rubygem-smart_proxy_pulp-1.3.0-1.el7.noarch
bacon-proxy.jomitsch.example.com-foreman-proxy-client-1.0-1.noarch
bacon-proxy.jomitsch.example.com-qpid-router-server-1.0-1.noarch
bacon-proxy.jomitsch.example.com-qpid-broker-1.0-1.noarch
foreman-proxy-1.18.0-0.develop.201802282028gitaf41016.el7.noarch
bacon-proxy.jomitsch.example.com-foreman-proxy-1.0-1.noarch
bacon-proxy.jomitsch.example.com-qpid-router-client-1.0-1.noarch
```

johnpmitsch replied:
We resolved this outside of this thread, misunderstanding on my part

mbacovsky replied:
@johnpmitsch, would it improve the readability if I renamed the installer_type symbols to `scenario`, `:legacy_katello` and `:legacy_capsule`. Or `:non_scenario_katello` and `:non_scenario_capsule`?

johnpmitsch replied:
@mbacovsky I think that would be a good idea. My vote goes to the `:legacy_` options

iNecas commented:
Could we leverage this code also in https://github.com/theforeman/foreman_maintain/blob/master/definitions/procedures/installer/upgrade.rb#L17, so that we could simplify the upgrade procedure?

mbacovsky commented:
@johnpmitsch, @iNecas, rebased, renamed the installer_type values, refactored the upgrade procedure. Changes were added in separate commit, feel free to squash before merge
Please test upgrade.

iNecas commented:
Thanks @mbacovsky and @ntkathole
